app/controllers/public_recipes_controller.rb

```diff
@@ -0,0 +1,11 @@
+class PublicRecipesController < ApplicationController
+  def index
+    @public_recipes = Recipe.where(public: true).includes(recipe_foods: :food_name)
+  end
```

Comment:
- [ ] 👏 Great progress on your recent changes! 👏 However, it appears that the N+1 problem still exists on this page. Resolving the N+1 issue is crucial for optimizing database queries and improving overall performance. Please review your code and address this problem to enhance the efficiency of your application.

![image](https://github.com/Kanyorok/TastyTracks/assets/55840999/1b474c25-4e58-4743-9baf-f1d393eea2f7)

This should help mitigate the N+1 problem and improve the performance of your page. Keep up the excellent work! 🚀📸.